src/test/java/net/minecraftforge/test/ShieldTest.java

```diff
@@ -0,0 +1,117 @@
+import net.minecraft.client.renderer.GlStateManager;
+import net.minecraft.client.renderer.block.model.ModelResourceLocation;
+import net.minecraft.client.renderer.tileentity.TileEntityRendererDispatcher;
+import net.minecraft.client.renderer.tileentity.TileEntitySpecialRenderer;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.entity.SharedMonsterAttributes;
+import net.minecraft.entity.ai.attributes.AttributeModifier;
+import net.minecraft.inventory.EntityEquipmentSlot;
+import net.minecraft.item.EnumDyeColor;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemShield;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.ItemSword;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.tileentity.TileEntityBanner;
+import net.minecraft.util.text.translation.I18n;
+import net.minecraftforge.client.ForgeHooksClient;
+import net.minecraftforge.client.model.ModelLoader;
+import net.minecraftforge.event.RegistryEvent;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.common.Mod.EventBusSubscriber;
+import net.minecraftforge.fml.common.event.FMLPreInitializationEvent;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+import net.minecraftforge.fml.common.registry.GameRegistry.ObjectHolder;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
+@ObjectHolder("shield_test")
+@EventBusSubscriber
+@Mod(modid = "shield_test", name = "Shield Test", version = "0.0.0", acceptableRemoteVersions = "*")
+public class ShieldTest
+{
+    public static final ItemShield DIAMOND_SHIELD = null;
+    public static final ItemSword HEAVY_DIAMOND_SWORD = null;
+    
+    private final ModelShield modelShield = new ModelShield();
+    
+    @SideOnly(Side.CLIENT)
+    @Mod.EventHandler
+    public void pre(FMLPreInitializationEvent event)
+    {
+        ModelLoader.setCustomModelResourceLocation(DIAMOND_SHIELD, 0, new ModelResourceLocation("minecraft:shield", "inventory"));
+        ModelLoader.setCustomModelResourceLocation(HEAVY_DIAMOND_SWORD, 0, new ModelResourceLocation("minecraft:diamond_sword", "inventory"));
+        TileEntityRendererDispatcher.instance.mapSpecialRenderers.put(TileEntityCustomShield.class, new TileEntitySpecialRenderer<TileEntityCustomShield>()
+                {
+            @Override
+            public void renderTileEntityAt(TileEntityCustomShield entity, double x, double y, double z, float partialTicks, int destroyStage)
+            {
+                Minecraft.getMinecraft().getTextureManager().bindTexture(BannerTextures.SHIELD_BASE_TEXTURE);
+                GlStateManager.pushMatrix();
+                GlStateManager.scale(1.0F, -1.0F, -1.0F);
+                modelShield.render();
+                GlStateManager.popMatrix();
+            }
+                });
+        ForgeHooksClient.registerTESRItemStack(DIAMOND_SHIELD, 0, TileEntityCustomShield.class);
```

Comment:
Then use a stick model as the shield. I don't want to endorse people making TESRs for ItemStacks for no reason.